### ISSUE TITLE
when a new window is created, the view is blurry on wayland with fractional scaling.

### ISSUE THREAD

**Issue by 3nt3** [qutebrowser/qutebrowser#6914]
**Version info**:

https://paste.the-compiler.org/view/19d8662c

**Does the bug happen if you start with `--temp-basedir`?**:
Yes.

**Description**
When a new window is created by executing `qutebrowser` instead of opening from an existing window (ctrl+n), the view is blurry on wayland with fractional scaling.

**How to reproduce**
1. Execute qutebrowser. 
2. Repeat 1

**Comment by 3nt3** [comment]
Update, new windows are always blurry now.

**Comment by The-Compiler** [comment]
Rendering is handled by the underlying QtWebEngine/Chromium, so I doubt there is much qutebrowser can do about this. Could you check whether you see the same issue with another project using QtWebEngine, such as KDE Falkon?

**Comment by The-Compiler** [comment]
@3nt3 any update on this?

**Comment by The-Compiler** [comment]
Closing due to inactivity.

**Comment by stacyharper** [comment]
Yes the issue is still present with Sway master. And no, falkon doesn't have the same issue.

![2024-01-09-09:51:52](https://github.com/qutebrowser/qutebrowser/assets/14213590/e2d54924-860b-4362-b69a-3d3fe8ed007a)

**Comment by agdestein** [comment]
This can be fixed by overriding the rounding policy to what used to be the default. When launching with
```sh
env QT_SCALE_FACTOR_ROUNDING_POLICY=round qutebrowser
```
everything renders chrystal clear.

**Comment by stacyharper** [comment]
Right thanks for the work-around

**Comment by The-Compiler** [comment]
Looks like this: [[QTBUG-113574] Incorrect sizing and bad text rendering with WebEngine using fractional scaling on Wayland - Qt Bug Tracker](https://bugreports.qt.io/browse/QTBUG-113574)

I suppose we could force that workaround on affected Qt versions until this is fixed upstream.

**Comment by fritzrehde** [comment]
I have this issue as well on Sway with fractional scaling 1.2. Unfortunately, @agdestein 's trick doesn't work for me. Interestingly, I didn't have these scaling issues for a long time, and only ran into them now after updating (though I hadn't updated in a while beforehand, so I don't know the specific version I was on before where the issue was not present).

**Comment by sysedwinistrator** [comment]
> Looks like this: [[QTBUG-113574] Incorrect sizing and bad text rendering with WebEngine using fractional scaling on Wayland - Qt Bug Tracker](https://bugreports.qt.io/browse/QTBUG-113574)

The bug has been fixed: https://codereview.qt-project.org/c/qt/qtwebengine/+/569910
The fix has already been cherry-picked into the upcoming 6.8 release branch: https://codereview.qt-project.org/c/qt/qtwebengine/+/573950
Qt 6.8 is due to be released end of September: https://wiki.qt.io/Qt_6.8_Release

I was also able to patch my version of qtwebengine (6.7.2 on nixpkgs/nixos-unstable) and rebuild qutebrowser against it. Fractional scaling now works correctly.

Maybe some distro package maintainers are willing to adopt the patch for qtwebengine in the meantime.
